sdk/storage/azure-storage-queues/inc/azure/storage/queues/queue_client.hpp

```diff
@@ -0,0 +1,126 @@
+  public:
+    static QueueClient CreateFromConnectionString(
+        const std::string& connectionString,
+        const std::string& queueName,
+        const QueueClientOptions& options = QueueClientOptions());
+
+    explicit QueueClient(
+        const std::string& queueUrl,
+        std::shared_ptr<StorageSharedKeyCredential> credential,
+        const QueueClientOptions& options = QueueClientOptions());
+
+    explicit QueueClient(
+        const std::string& queueUrl,
+        std::shared_ptr<Core::Credentials::TokenCredential> credential,
+        const QueueClientOptions& options = QueueClientOptions());
+
+    explicit QueueClient(
+        const std::string& queueUrl,
+        const QueueClientOptions& options = QueueClientOptions());
+
+    std::string GetUrl() const { return m_queueUrl.GetAbsoluteUrl(); }
+
+    Azure::Response<Models::CreateQueueResult> Create(
+        const CreateQueueOptions& options = CreateQueueOptions(),
+        const Azure::Core::Context& context = Azure::Core::Context()) const;
+
+    Azure::Response<Models::CreateQueueResult> CreateIfNotExists(
+        const CreateQueueOptions& options = CreateQueueOptions(),
+        const Azure::Core::Context& context = Azure::Core::Context()) const;
+
+    Azure::Response<Models::DeleteQueueResult> Delete(
+        const DeleteQueueOptions& options = DeleteQueueOptions(),
+        const Azure::Core::Context& context = Azure::Core::Context()) const;
+
+    Azure::Response<Models::DeleteQueueResult> DeleteIfExists(
+        const DeleteQueueOptions& options = DeleteQueueOptions(),
+        const Azure::Core::Context& context = Azure::Core::Context()) const;
+
+    Azure::Response<Models::QueueProperties> GetProperties(
+        const GetQueuePropertiesOptions& options = GetQueuePropertiesOptions(),
+        const Azure::Core::Context& context = Azure::Core::Context()) const;
+
+    Azure::Response<Models::SetQueueMetadataResult> SetMetadata(
+        Metadata metadata,
+        const SetQueueMetadataOptions& options = SetQueueMetadataOptions(),
+        const Azure::Core::Context& context = Azure::Core::Context()) const;
+
+    Azure::Response<Models::QueueAccessPolicy> GetAccessPolicy(
+        const GetQueueAccessPolicyOptions& options = GetQueueAccessPolicyOptions(),
+        const Azure::Core::Context& context = Azure::Core::Context()) const;
+
+    Azure::Response<Models::SetQueueAccessPolicyResult> SetAccessPolicy(
+        std::vector<Models::SignedIdentifier> signedIdentifiers,
+        const SetQueueAccessPolicyOptions& options = SetQueueAccessPolicyOptions(),
+        const Azure::Core::Context& context = Azure::Core::Context()) const;
+
+    Azure::Response<Models::SendMessageResult> SendMessage(
+        std::string messageText,
+        const SendMessageOptions& options = SendMessageOptions(),
+        const Azure::Core::Context& context = Azure::Core::Context()) const;
+
+    Azure::Response<Models::QueueMessage> ReceiveMessage(
```

Comment:
GetXMethod should return X rather than GetXResult. This is agreed by C++ working group. Ref to the email titled _[SDK model type naming consistency] Client library GetX Methods should return model types named X and not GetXResult_ by Ahson.

Reply:
This is not a `GetXMethod` is it?  This is a `ReceiveXMethod`  .  Would the rule cover all `verbXMethod` like
- DeleteX
- PutX
- SendX
- LeaseX
- etcX
??

Reply:
discussing via email